From 139efe758c358327262ddb235f0043562ff1e39f Mon Sep 17 00:00:00 2001
From: Stefan Bader <stefan.bader@canonical.com>
Date: Tue, 25 Mar 2014 14:27:08 +0100
Subject: [PATCH] libxl: Create log directory earlier

Commit d9f19c30d054c86b15a304f4118baa4fa75af9d2 moved a lot of the
configuration setup into libxlDriverConfigNew().
However that tries to create the libxl/libxl-driver.log before the
libxl directory gets created in libxlStateInitialize().

This causes the daemon to fail on systems that have not had the directory
created before.

Move the code to create the libxl directory into libxlDriverConfigNew().

(cherry-picked from 139efe758c358327262ddb235f0043562ff1e39f upstream)
Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
---
 src/libxl/libxl_conf.c   |    8 ++++++++
 src/libxl/libxl_driver.c |    7 -------
 2 files changed, 8 insertions(+), 7 deletions(-)

Index: libvirt-1.2.2/src/libxl/libxl_conf.c
===================================================================
--- libvirt-1.2.2.orig/src/libxl/libxl_conf.c	2014-03-25 14:40:04.000000000 +0100
+++ libvirt-1.2.2/src/libxl/libxl_conf.c	2014-03-25 17:03:41.457521722 +0100
@@ -1085,6 +1085,14 @@ libxlDriverConfigNew(void)
     if (virAsprintf(&log_file, "%s/libxl-driver.log", cfg->logDir) < 0)
         goto error;
 
+    if (virFileMakePath(cfg->logDir) < 0) {
+        virReportError(VIR_ERR_INTERNAL_ERROR,
+                       _("failed to create log dir '%s': %s"),
+                       cfg->logDir,
+                       virStrerror(errno, ebuf, sizeof(ebuf)));
+        goto error;
+    }
+
     if ((cfg->logger_file = fopen(log_file, "a")) == NULL)  {
         VIR_ERROR(_("Failed to create log file '%s': %s"),
                   log_file, virStrerror(errno, ebuf, sizeof(ebuf)));
Index: libvirt-1.2.2/src/libxl/libxl_driver.c
===================================================================
--- libvirt-1.2.2.orig/src/libxl/libxl_driver.c	2014-03-25 14:40:04.000000000 +0100
+++ libvirt-1.2.2/src/libxl/libxl_driver.c	2014-03-25 17:03:41.461521736 +0100
@@ -1032,13 +1032,6 @@ libxlStateInitialize(bool privileged,
         goto error;
 
     libxl_driver->config = cfg;
-    if (virFileMakePath(cfg->logDir) < 0) {
-        virReportError(VIR_ERR_INTERNAL_ERROR,
-                       _("failed to create log dir '%s': %s"),
-                       cfg->logDir,
-                       virStrerror(errno, ebuf, sizeof(ebuf)));
-        goto error;
-    }
     if (virFileMakePath(cfg->stateDir) < 0) {
         virReportError(VIR_ERR_INTERNAL_ERROR,
                        _("failed to create state dir '%s': %s"),
